src/libcore/iter/mod.rs

```diff
@@ -643,6 +643,43 @@ impl<I> Iterator for Cycle<I> where I: Clone + Iterator {
             _ => (usize::MAX, None)
         }
     }
+
+    #[inline]
+    fn all<F>(&mut self, f: F) -> bool where F: FnMut(Self::Item) -> bool {
+        self.iter.clone().chain(self.orig.clone()).all(f)
```

Comment:
If this short-circuits with an element inside `self.orig`, this iterator should continue with the next element after the matching one in `self.orig`. Currently it'll instead restart with the first element in `self.orig`.

Reply:
Ah, that's a good point. I'll fix that if the decision is to merge.